File: app/src/main/java/com/fractalwrench/crazycats/mocks/TestDependencies.java
package com.fractalwrench.crazycats.mocks;


import com.fractalwrench.crazycats.injection.DefaultSchedulers;

import java.io.IOException;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

public class TestDependencies {

    private static DefaultSchedulers schedulers() {
        Scheduler immediate = Schedulers.trampoline();
        return new DefaultSchedulers(immediate, immediate);
    }

    public static MockSuccessRepository mockSuccessRepository() throws IOException {
        return new MockSuccessRepository(schedulers());
    }

    public static MockFailureRepository mockFailureRepository() throws IOException {
        return new MockFailureRepository();
    }

}

File: app/src/main/java/com/fractalwrench/crazycats/mocks/TestDependencies.kt
package com.fractalwrench.crazycats.mocks


import com.fractalwrench.crazycats.injection.DefaultSchedulers
import io.reactivex.schedulers.Schedulers
import java.io.IOException

object TestDependencies {

    private fun schedulers(): DefaultSchedulers {
        val immediate = Schedulers.trampoline()
        return DefaultSchedulers(immediate, immediate)
    }

    @Throws(IOException::class)
    fun mockSuccessRepository(): MockSuccessRepository {
        return MockSuccessRepository(schedulers())
    }

    @Throws(IOException::class)
    fun mockFailureRepository(): MockFailureRepository {
        return MockFailureRepository()
    }

}
